### PR TITLE
add an example of programatically creating a named tuple

### PR DESCRIPTION
**Pull request by KristofferC**
This seems to be relatively unknown.

### CODE REVIEW

File: base/namedtuple.jl [hunk 1]
```diff
@@ -44,6 +44,16 @@ julia> collect(pairs(x))
  :a => 1
  :b => 2
 ```
+
+A named tuple can be created programatically by splatting an iterator yielding
```
Comment (dkarrasch) [inline base/namedtuple.jl]: programmatically? I'm not a native speaker, but I couldn't find programatically with a single M.

https://www.merriam-webster.com/dictionary/programmatically

**Comment by pshashk** [comment]
Version without comma works as well `(;zip(keys, values)...)`

**Comment by KristofferC** [comment]
Thanks for comments, updated accordingly.

File: base/namedtuple.jl [hunk 2]
```diff
@@ -44,6 +44,16 @@ julia> collect(pairs(x))
...
+
+A named tuple can be created programmatically by splatting an iterator yielding
+`name => value` pairs where each `name` is a `Symbol`:
+
+```jldoctest
+julia> keys = (:a, :b, :c); values = (1, 2, 3);
+
+julia> (; zip(keys, values)...)
+(a = 1, b = 2, c = 3)
+```
```
Comment (JeffBezanson) [inline base/namedtuple.jl]: :+1: While we're at it, can also mention the syntax `(; :a => 1)` (where the name is computed for one element at a time).

Comment (mbauman) [reply base/namedtuple.jl]: Perhaps explicitly call out the commonality to calling functions with keyword arguments?

Comment (KristofferC) [reply base/namedtuple.jl]: Updated.

**Comment by o314** [comment]
I was just going to propose a patch for this. It's not on v1.2
But you've done it for v1.3 (or v1.4)
Thanks.


```
!!! NamedTuple `nt` can be reconstruct from iterator `itr` yielded by `pairs(nt)`
with the syntax `(;itr...)`

...

julia> itr = pairs((a = 1, b = 2)); (;itr...)
(a = 1, b = 2)
```

**Pull request merged**